### PR TITLE
docs: add README and HACKING documentation

### PR DESCRIPTION
**Pull request by rphillips**

### CODE REVIEW

File: HACKING.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+operator-sdk build quay.io/[some_username]/pod-checkpointer-operator
+docker push quay.io/[some_username]/pod-checkpointer-operator
+```
+
+### Running
+
+To run the operator, first deploy the custom resource definitions:
+
+```
+for f in manifests/*.yaml; do oc create -f $f ; done
```
Comment (derekwaynecarr) [inline HACKING.md]: should just be able to apply a directory?

Comment (rphillips) [reply HACKING.md]: thanks. fixed in feb1bdd

File: HACKING.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+* An OpenShift or Kubernetes cluster with at least a master, infra, and compute
+node. An admin-scoped * `KUBECONFIG` for the cluster. The [operator-sdk](https://github.com/operator-framework/operator-sdk).
+
+### Building
+
+To build the operator during development, use the standard Go toolchain:
+
+```
+$ go build ./...
```
Comment (sjenning) [inline HACKING.md]: We have a makefile.  Maybe change this to `make build`.

File: HACKING.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+```
+
+### Container image
+
+To build the operator docker image:
```
Comment (sjenning) [inline HACKING.md:22]: `make images`

File: HACKING.md (hunk 2 above)
Comment (rphillips) [reply HACKING.md]: fixed in 9495d2b

File: HACKING.md (hunk 3 above)
Comment (rphillips) [reply HACKING.md:22]: fixed in df1570f

**Pull request merged**